main.cpp

```diff
@@ -275,296 +300,283 @@ struct Bakery
+}
 
 struct AquaPark
 {
-    //number of pools (int)
     int numPools = 5;
-    //number of saunas (int)
     int numSaunas = 3;
-    //total length of water slides (double)
     float totalWaterSlidesLength = 121.56f;
-    //opening hours (float)
     float openAt = 9.30f;
-    //ticket price (float)
     float regularTicker = 13.21f;
 
-    //relax muscles
     void relaxMuscles(bool goToSauna, int swimmingIntensity);
-    //be fun
     void haveFun(float totalWaterSlidesLength, int numberOfSlides);
-    //learning swimming
     bool learnToSwim(int age, int timeSpent);
 };
 
-/*
-Thing 4) house
-5 properties:
-    1) number of windows (int)
-    2) size of living room (float)
-    3) room height (float)
-    4) number of bathrooms (int)
-    5) total floor size (float)
-3 things it can do:
-    1) provide shelter
-    2) provide rest
-    3) get dirty
- */
+void AquaPark::relaxMuscles(bool goToSauna, int swimmingIntensity)
+{
+    goToSauna = false;
+    if (swimmingIntensity > 5)
+        goToSauna = true;
+        std::cout << "You should use one of the sauna's to relax your muscles!" << std::endl;
+}
+void AquaPark::haveFun(float aTotalWaterSlidesLength, int numberOfSlides)
+{
+    std::cout << "Total length of our " << numberOfSlides << " slides is: " << aTotalWaterSlidesLength << " which means that it's not possible to NOT have fun!" << std::endl;
+}
+bool AquaPark::learnToSwim(int age, int timeSpent)
+{
+    if (age < 10 && timeSpent > 60)
+        return true;
+    else if (age >= 10 && timeSpent > 120)
+        return true;
+    else
+        return false;
+
+}
 
 struct House
 {
-    //number of windows (int)
     int numWindows = 8;
-    //size of living room (float)
     float livingRoomSize = 31.8f;
-    //room height (float)
     float roomHeight = 2.8f;
-    //number of bathrooms (int)
     int numBathRooms = 3;
-    //total floor size (float)
     float totalFloorSize = 155.2f;
 
-    //provide shelter
     void provideShelter();
-    //provide rest
     void provideRest(float sleepQuality);
-    //get dirty
-    bool getDirty(int numPeople, int numAnimals, float totalFloorSize);
+    bool getDirty(int numPeople, int numAnimals, float hTotalFloorSize);
 };
 
- /*
-Thing 5) vca
-5 properties:
-    1) minimum gain (double)
-    2) maximum gain (double)
-    3) initial gain (double)
-    4) modulation attenuator (float)
-    5) inputs (int)
-3 things it can do:
-    1) set inital gain
-    2) add rhythm to signal
-    3) silence the signal
- */
+void House::provideShelter()
+{
+}
+
+void House::provideRest(float sleepQuality)
+{
+    if (sleepQuality > 3)
+        std::cout << "Conratulations, you managed to get rest!" << std::endl;
+    else
+        std::cout << "Sorry buddy, you'll be sleepy for the rest of the day." << std::endl;
+}
+
+bool House::getDirty(int numPeople, int numAnimals, float hTotalFloorSize)
+{
+    if (hTotalFloorSize / (numPeople+numAnimals) > 20)
+        return true;
+    else
+        return false;
+}
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
You do not need the `else` block.  Here's why:
`return` statements are the last thing that **ever happens** in a function.
When the `if()` expression is evaluated as `true` and the block is entered, the `return aValue;` is the last line of code to be executed before the function exits.
but when the `if()` expression evaluates as `false` and the block is **NOT** entered, the next thing that happens is `return anotherValue;`
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
that means you do **NOT** need the `else` block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    
    return anotherValue;
}
```
- [ ] revise